### PR TITLE
Fix #23557 - Ignore types with DW_AT_declaration in DWARF

### LINKED ISSUE #23557: sdb types are overwritten from other compilation units with same name
## Environment

```sh
ter 29 out 2024 11:29:20 -03
radare2 5.9.4 0 @ linux-x86-64
birth: git.5.9.4 2024-09-28__12:07:37
options: gpl release -O1 cs:5 cl:2 meson
Linux x86_64
```

## Description

DWARF type structures that are incomplete (DW_AT_declaration) are inserted in sdb_types overwriting the concrete type that was previously added, making radare2 lose track of all struct members. There's another related issue here that even if the type is concrete but not exported in a header file, it will overwrite one from another compilation unit with the same name, because types doesn't distinguish from compilation unit atm.

## Test

https://github.com/radareorg/radare2-testbins/pull/98

be aware that the order of the arguments in gcc makes DW_AT_declaration appear first then it won't reproduce

### PR DESCRIPTION
**Pull request by wgrr**
- [x] Mark this if you consider it ready to merge
- [x] I've added tests (optional)

### CODE REVIEW

File: libr/anal/type.c [hunk 1]
```diff
@@ -385,7 +385,13 @@ static void save_struct(const RAnal *anal, const RAnalBaseType *type) {
 	}
 	// struct.name=param1,param2,paramN
 	char *key = r_str_newf ("%s.%s", kind, sname);
-	sdb_set_owned (anal->sdb_types, key, r_strbuf_drain (arglist), 0);
+	if (sdb_exists (anal->sdb_types, key)) {
+		R_LOG_WARN ("Ignoring overwrite of type \"%s\" in sdb_types.", key);
```
Comment (trufae) [inline libr/anal/type.c:389]:
```suggestion
		R_LOG_WARN ("Ignoring overwrite of type \"%s\" in sdb_types", key);
```

**Review by trufae — CHANGES_REQUESTED** [review]
^

**Comment by wgrr** [comment]
I've just noticed that I forgot to update the file name in cmd_tc test after I changed it to use underscore like other execs in bins, done

File: libr/anal/type.c [hunk 2]
```diff
@@ -385,7 +385,13 @@ static void save_struct(const RAnal *anal, const RAnalBaseType *type) {
 	}
 	// struct.name=param1,param2,paramN
 	char *key = r_str_newf ("%s.%s", kind, sname);
-	sdb_set_owned (anal->sdb_types, key, r_strbuf_drain (arglist), 0);
+	if (sdb_exists (anal->sdb_types, key)) {
+		R_LOG_WARN ("Ignoring overwrite of type '%s' in sdb_types.", key);
```
Comment (trufae) [inline libr/anal/type.c:389]:
```suggestion
		R_LOG_WARN ("Ignoring overwrite of type '%s' in sdb_types", key);
```

Comment (trufae) [reply libr/anal/type.c:389]: remove the dot please. otherwise the linter will fail

**Comment by trufae** [comment]
perfect 👌

**Pull request merged**